forge-gui/res/cardsfolder/upcoming/vault_112_sadistic_simulation.txt

```diff
@@ -0,0 +1,14 @@
+Name:Vault 112: Sadistic Simulation
+ManaCost:2 U R
+Types:Enchantment Saga
+K:Chapter:3:DBTap,DBTap,DBChooseNumber
+SVar:DBTap:DB$ Tap | ValidTgts$ Creature | SubAbility$ DBCounter | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature | SpellDescription$ Tap up to one target creature and put a stun counter on it. You get {E}{E} (two energy counters).
+SVar:DBCounter:DB$ PutCounter | Defined$ Targeted | CounterType$ Stun | CounterNum$ 1 | SubAbility$ DBEnergy
+SVar:DBEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ 2
+SVar:DBChooseNumber:DB$ ChooseNumber | Max$ Max | ListTitle$ Pay Energy for ability| SubAbility$ DBExile | SpellDescription$ Pay any amount of {E}. If you paid one or more {E} this way, shuffle your library, then exile that many cards from the top. You may play one of those cards without paying its mana cost.
+SVar:DBExile:DB$ Dig | UnlessCost$ PayEnergy<X> | UnlessPayer$ You | UnlessSwitched$ True | Defined$ You | DigNum$ X | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBPlay
```

Comment:
shuffle missing

Reply:
did you test what happens when you pay 0 energy?

Reply:
Nothing happend. Is this correct ?

Reply:
then why does _Cosi's Trickster_ trigger even if I pay 0?

Reply:
Hmm is there a param to prevent paying 1 ?

Reply:
1. your effect order is still wrong
2. the way it's worded paying 0 can kind of be done
3. but checking against ChosenNumber might be acceptable here

Reply:
But what if you choose 1 but pay nothing

Reply:
check _Greenbelt Rampager_